libtrap/src/trap.c

```diff
@@ -3169,50 +3098,59 @@ int trap_ctx_get_data_fmt(trap_ctx_t *ctx, uint8_t ifc_dir, uint32_t ifc_idx, ui
    trap_ctx_priv_t *c = ctx;
 
    if (ctx == NULL) {
-      return TRAP_E_NOT_INITIALIZED;
+      return trap_error(ctx, TRAP_E_NOT_INITIALIZED);
    }
 
    if (ifc_dir == TRAPIFC_INPUT) {
       if (ifc_idx >= c->num_ifc_in) {
-         return TRAP_E_BAD_IFC_INDEX;
+         return trap_error(ctx, TRAP_E_BAD_IFC_INDEX);
       }
 
       inifc = &c->in_ifc_list[ifc_idx];
 
+      pthread_mutex_lock(&inifc->ifc_mtx);
       if (inifc->data_type == TRAP_FMT_RAW) {
-         return TRAP_E_BADPARAMS;
+         pthread_mutex_unlock(&inifc->ifc_mtx);
+         return trap_error(ctx, TRAP_E_BADPARAMS);
```

Comment:
Not related to the changes you made, but ... why BADPARAMS should be returned here? I don't understand why is it forbidden to call this function with RAW data type. It should return `TRAP_FMT_RAW` as `data_type` and `NULL` as `spec` (below is the code which would do exactly this, but line 3123 is currently dead, since the function returns here).

Reply:
Will be addressed in separate issue.

Reply:
Yes, that makes sense, thanks for the fix.